core/Plugin.php

```diff
@@ -451,6 +451,19 @@ public static function getPluginNameFromNamespace($namespaceOrClassName)
     }
 
     /**
+     * Override this method in your plugin class if you want your plugin to be loaded during tracking.
+     *
+     * Note: If you define your own dimension or handle a tracker event, your plugin will automatically
+     * be detected as a tracker plugin.
+     *
+     * @return bool
+     */
+    public function isTrackerPlugin()
```

Comment:
I reckon it will be good to talk about this in Poland. [Convention over configuration](https://en.wikipedia.org/wiki/Convention_over_configuration) is partially dictating where classes have to be but that's wanted re simplicity etc. It is easier to document, having only one option to do something, it is easy to find things in any plugin etc. 

From Wikipedia https://en.wikipedia.org/wiki/Convention_over_configuration: 
> Convention over configuration (also known as coding by convention) is a software design paradigm which seeks to decrease the number of decisions that developers need to make, gaining simplicity, and not necessarily losing flexibility.

DI is more important for some core parts, like being able to replace DB (at some point), or Cache, or Filesystem, or ... but not really for our plugins.